src/shims/fs.rs

```diff
@@ -305,16 +305,27 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let path: PathBuf = this.read_os_str_from_c_str(pathname_scalar)?.into();
         // `flags` should be a `c_int` but the `syscall` function provides an `isize`.
-        let flags: i32 = this.try_from_ptr_sized_operand(flags_op)?;
-
+        let flags: i32 = this
+            .read_scalar(flags_op)?
+            .to_machine_isize(&*this.tcx)?
+            .try_into()
+            .map_err(|e| err_unsup_format!(
+                "Failed to convert pointer sized operand to integer: {}",
+                e
+            ))?;
+        // `dirfd` should be a `c_int` but the `syscall` function provides an `isize`.
+        let dirfd: i32 = this
+            .read_scalar(dirfd_op)?
+            .to_machine_isize(&*this.tcx)?
+            .try_into()
+            .map_err(|e| err_unsup_format!(
+                "Failed to convert pointer sized operand to integer: {}",
+                e
+            ))?;
         // we only support interpreting `path` as an absolute directory or as a directory relative
         // to `dirfd` when the latter is `AT_FDCWD`. Other behavior is specified but it cannot be
         // tested from `libstd`. If you found this error, please open an issue reporting it.
-
-        // `dirfd` should be a `c_int` but the `syscall` function provides an `isize`.
-        if !(path.is_absolute()
-            || this.try_from_ptr_sized_operand::<i32>(dirfd_op)?
-                == this.eval_libc_i32("AT_FDCWD")?)
+        if !(path.is_absolute() || dirfd == this.eval_libc_i32("AT_FDCWD")?)
         {
             throw_unsup_format!(
                 "Using statx with a file descriptor different from `AT_FDCWD` is not supported"
```

Comment:
Also mention the relative path condition. "statx with a relative path and a directory file descriptor other than `AT_FDCWD`", or so.

Reply:
The message seems unchanged? Did you forget to push?

Reply:
Oh shooo my bad. I thought you meant the comment, not the error. I'll change it